STM32F1/cores/maple/Print.cpp

```diff
@@ -208,24 +208,72 @@ size_t Print::println(const Printable& x)
   return n;
 }
 
-#ifdef SUPPORTS_PRINTF
+
+//#ifdef SUPPORTS_PRINTF
+//Note: ifdef not need - not used printf is not linked
+
+// method uses the GLIBS extension fopencookie (), which works on the platforms STM32, SAMD, ESP8266, Linux ..
+// And work with any devices: Serial, TFT, Flash &etc. 
+//( For avr-libc, a similar functional is implemented through specific properties of type FILE, without any functions.)
+
+#define _GNU_SOURCE       
 #include <stdio.h>
 #include <stdarg.h>
-// Work in progress to support printf.
-// Need to implement stream FILE to write individual chars to chosen serial port
-int Print::printf (__const char *__restrict __format, ...)
- {
-FILE *__restrict __stream;
-     int ret_status = 0;
-
-
-     va_list args;
-     va_start(args,__format);
-     ret_status = vfprintf(__stream, __format, args);
-     va_end(args);
-     return ret_status;
- }
- #endif
+
+/*  
+//Bug detected 
+//failed: fopencookie( 0, "w+",  { NULL,  WR_fn, NULL, NULL }) 
+//failed: fopencookie( 0, "rw+", { NULL, WR_fn, NULL, NULL }) 
+//
+//worked: fopencookie( 0, "rw+", { RD_fn, WR_fn, NULL, NULL }) 
+//worked: fopencookie( 0, "rw+", { NULL,  WR_fn, NULL, NULL })
```

Comment:
Line 229 (worked) seems to be identical to line 226, which failed.
Btw, do we really need this comment block? I would remove it.
The solution matters, which is implemented below.

Reply:
Block comments should be removed. :)

The GLIBC documentation says that when fopencookie () is called, pointers to both the write () and read () functions must be passed.

That is, if reading is not required, a valid pointer to a dummy function should be passed, and not NULL. Otherwise it does not work.

Reply:
I understand the reason, I just asked to remove unneeded comment block (which includes contradictory information).